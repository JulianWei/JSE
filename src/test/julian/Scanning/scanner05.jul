module m1;
import m2;
import m3.F as F;

enum E{}

attribute A{}

public abstract class C {
	private static const int a = 2;
	protected bool b;
	public string c;
	private C(){};
	public C(int a) : this(){}
	public C(int a, var b, char c, string s, float f) : super(a){}
	public void fm1(){}
	abstract void fm0();
	public hosted fh();
	int fm2(int x){
		bool b = x is int;
		try {
			while(b){
				do{
					break;
				} while (false);
				continue;
			}			
		} 
		catch (Exception e) { } 
		finally { }
		return 1;
	}
}