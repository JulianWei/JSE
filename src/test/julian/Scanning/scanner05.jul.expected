[MODULE]	module		index: 0 - 5 line/column: 0:0 - 0:5
[IDENTIFIER]	m1		index: 7 - 8 line/column: 0:7 - 0:8
[SEMICOLON]	;		index: 9 - 9 line/column: 0:9 - 0:9
[IMPORT]	import		index: 11 - 16 line/column: 1:0 - 1:5
[IDENTIFIER]	m2		index: 18 - 19 line/column: 1:7 - 1:8
[SEMICOLON]	;		index: 20 - 20 line/column: 1:9 - 1:9
[IMPORT]	import		index: 22 - 27 line/column: 2:0 - 2:5
[IDENTIFIER]	m3		index: 29 - 30 line/column: 2:7 - 2:8
[DOT]	.		index: 31 - 31 line/column: 2:9 - 2:9
[IDENTIFIER]	F		index: 32 - 32 line/column: 2:10 - 2:10
[AS]	as		index: 34 - 35 line/column: 2:12 - 2:13
[IDENTIFIER]	F		index: 37 - 37 line/column: 2:15 - 2:15
[SEMICOLON]	;		index: 38 - 38 line/column: 2:16 - 2:16
[ENUM]	enum		index: 41 - 44 line/column: 4:0 - 4:3
[IDENTIFIER]	E		index: 46 - 46 line/column: 4:5 - 4:5
[LEFT_CURLY]	{		index: 47 - 47 line/column: 4:6 - 4:6
[RIGHT_CURLY]	}		index: 48 - 48 line/column: 4:7 - 4:7
[ATTRIBUTE]	attribute		index: 51 - 59 line/column: 6:0 - 6:8
[IDENTIFIER]	A		index: 61 - 61 line/column: 6:10 - 6:10
[LEFT_CURLY]	{		index: 62 - 62 line/column: 6:11 - 6:11
[RIGHT_CURLY]	}		index: 63 - 63 line/column: 6:12 - 6:12
[PUBLIC]	public		index: 66 - 71 line/column: 8:0 - 8:5
[ABSTRACT]	abstract		index: 73 - 80 line/column: 8:7 - 8:14
[CLASS]	class		index: 82 - 86 line/column: 8:16 - 8:20
[IDENTIFIER]	C		index: 88 - 88 line/column: 8:22 - 8:22
[LEFT_CURLY]	{		index: 90 - 90 line/column: 8:24 - 8:24
[PRIVATE]	private		index: 93 - 99 line/column: 9:1 - 9:7
[STATIC]	static		index: 101 - 106 line/column: 9:9 - 9:14
[CONST]	const		index: 108 - 112 line/column: 9:16 - 9:20
[INT]	int		index: 114 - 116 line/column: 9:22 - 9:24
[IDENTIFIER]	a		index: 118 - 118 line/column: 9:26 - 9:26
[ASSIGN]	=		index: 120 - 120 line/column: 9:28 - 9:28
[NUMBER]	2		index: 122 - 122 line/column: 9:30 - 9:30
[SEMICOLON]	;		index: 123 - 123 line/column: 9:31 - 9:31
[PROTECTED]	protected		index: 126 - 134 line/column: 10:1 - 10:9
[BOOL]	bool		index: 136 - 139 line/column: 10:11 - 10:14
[IDENTIFIER]	b		index: 141 - 141 line/column: 10:16 - 10:16
[SEMICOLON]	;		index: 142 - 142 line/column: 10:17 - 10:17
[PUBLIC]	public		index: 145 - 150 line/column: 11:1 - 11:6
[STRING]	string		index: 152 - 157 line/column: 11:8 - 11:13
[IDENTIFIER]	c		index: 159 - 159 line/column: 11:15 - 11:15
[SEMICOLON]	;		index: 160 - 160 line/column: 11:16 - 11:16
[PRIVATE]	private		index: 163 - 169 line/column: 12:1 - 12:7
[IDENTIFIER]	C		index: 171 - 171 line/column: 12:9 - 12:9
[LEFT_PAREN]	(		index: 172 - 172 line/column: 12:10 - 12:10
[RIGHT_PAREN]	)		index: 173 - 173 line/column: 12:11 - 12:11
[LEFT_CURLY]	{		index: 174 - 174 line/column: 12:12 - 12:12
[RIGHT_CURLY]	}		index: 175 - 175 line/column: 12:13 - 12:13
[SEMICOLON]	;		index: 176 - 176 line/column: 12:14 - 12:14
[PUBLIC]	public		index: 179 - 184 line/column: 13:1 - 13:6
[IDENTIFIER]	C		index: 186 - 186 line/column: 13:8 - 13:8
[LEFT_PAREN]	(		index: 187 - 187 line/column: 13:9 - 13:9
[INT]	int		index: 188 - 190 line/column: 13:10 - 13:12
[IDENTIFIER]	a		index: 192 - 192 line/column: 13:14 - 13:14
[RIGHT_PAREN]	)		index: 193 - 193 line/column: 13:15 - 13:15
[COLON]	:		index: 195 - 195 line/column: 13:17 - 13:17
[THIS]	this		index: 197 - 200 line/column: 13:19 - 13:22
[LEFT_PAREN]	(		index: 201 - 201 line/column: 13:23 - 13:23
[RIGHT_PAREN]	)		index: 202 - 202 line/column: 13:24 - 13:24
[LEFT_CURLY]	{		index: 203 - 203 line/column: 13:25 - 13:25
[RIGHT_CURLY]	}		index: 204 - 204 line/column: 13:26 - 13:26
[PUBLIC]	public		index: 207 - 212 line/column: 14:1 - 14:6
[IDENTIFIER]	C		index: 214 - 214 line/column: 14:8 - 14:8
[LEFT_PAREN]	(		index: 215 - 215 line/column: 14:9 - 14:9
[INT]	int		index: 216 - 218 line/column: 14:10 - 14:12
[IDENTIFIER]	a		index: 220 - 220 line/column: 14:14 - 14:14
[COMMA]	,		index: 221 - 221 line/column: 14:15 - 14:15
[VAR]	var		index: 223 - 225 line/column: 14:17 - 14:19
[IDENTIFIER]	b		index: 227 - 227 line/column: 14:21 - 14:21
[COMMA]	,		index: 228 - 228 line/column: 14:22 - 14:22
[CHAR]	char		index: 230 - 233 line/column: 14:24 - 14:27
[IDENTIFIER]	c		index: 235 - 235 line/column: 14:29 - 14:29
[COMMA]	,		index: 236 - 236 line/column: 14:30 - 14:30
[STRING]	string		index: 238 - 243 line/column: 14:32 - 14:37
[IDENTIFIER]	s		index: 245 - 245 line/column: 14:39 - 14:39
[COMMA]	,		index: 246 - 246 line/column: 14:40 - 14:40
[FLOAT]	float		index: 248 - 252 line/column: 14:42 - 14:46
[IDENTIFIER]	f		index: 254 - 254 line/column: 14:48 - 14:48
[RIGHT_PAREN]	)		index: 255 - 255 line/column: 14:49 - 14:49
[COLON]	:		index: 257 - 257 line/column: 14:51 - 14:51
[SUPER]	super		index: 259 - 263 line/column: 14:53 - 14:57
[LEFT_PAREN]	(		index: 264 - 264 line/column: 14:58 - 14:58
[IDENTIFIER]	a		index: 265 - 265 line/column: 14:59 - 14:59
[RIGHT_PAREN]	)		index: 266 - 266 line/column: 14:60 - 14:60
[LEFT_CURLY]	{		index: 267 - 267 line/column: 14:61 - 14:61
[RIGHT_CURLY]	}		index: 268 - 268 line/column: 14:62 - 14:62
[PUBLIC]	public		index: 271 - 276 line/column: 15:1 - 15:6
[VOID]	void		index: 278 - 281 line/column: 15:8 - 15:11
[IDENTIFIER]	fm1		index: 283 - 285 line/column: 15:13 - 15:15
[LEFT_PAREN]	(		index: 286 - 286 line/column: 15:16 - 15:16
[RIGHT_PAREN]	)		index: 287 - 287 line/column: 15:17 - 15:17
[LEFT_CURLY]	{		index: 288 - 288 line/column: 15:18 - 15:18
[RIGHT_CURLY]	}		index: 289 - 289 line/column: 15:19 - 15:19
[ABSTRACT]	abstract		index: 292 - 299 line/column: 16:1 - 16:8
[VOID]	void		index: 301 - 304 line/column: 16:10 - 16:13
[IDENTIFIER]	fm0		index: 306 - 308 line/column: 16:15 - 16:17
[LEFT_PAREN]	(		index: 309 - 309 line/column: 16:18 - 16:18
[RIGHT_PAREN]	)		index: 310 - 310 line/column: 16:19 - 16:19
[SEMICOLON]	;		index: 311 - 311 line/column: 16:20 - 16:20
[PUBLIC]	public		index: 314 - 319 line/column: 17:1 - 17:6
[HOSTED]	hosted		index: 321 - 326 line/column: 17:8 - 17:13
[IDENTIFIER]	fh		index: 328 - 329 line/column: 17:15 - 17:16
[LEFT_PAREN]	(		index: 330 - 330 line/column: 17:17 - 17:17
[RIGHT_PAREN]	)		index: 331 - 331 line/column: 17:18 - 17:18
[SEMICOLON]	;		index: 332 - 332 line/column: 17:19 - 17:19
[INT]	int		index: 335 - 337 line/column: 18:1 - 18:3
[IDENTIFIER]	fm2		index: 339 - 341 line/column: 18:5 - 18:7
[LEFT_PAREN]	(		index: 342 - 342 line/column: 18:8 - 18:8
[INT]	int		index: 343 - 345 line/column: 18:9 - 18:11
[IDENTIFIER]	x		index: 347 - 347 line/column: 18:13 - 18:13
[RIGHT_PAREN]	)		index: 348 - 348 line/column: 18:14 - 18:14
[LEFT_CURLY]	{		index: 349 - 349 line/column: 18:15 - 18:15
[BOOL]	bool		index: 353 - 356 line/column: 19:2 - 19:5
[IDENTIFIER]	b		index: 358 - 358 line/column: 19:7 - 19:7
[ASSIGN]	=		index: 360 - 360 line/column: 19:9 - 19:9
[IDENTIFIER]	x		index: 362 - 362 line/column: 19:11 - 19:11
[IS]	is		index: 364 - 365 line/column: 19:13 - 19:14
[INT]	int		index: 367 - 369 line/column: 19:16 - 19:18
[SEMICOLON]	;		index: 370 - 370 line/column: 19:19 - 19:19
[TRY]	try		index: 374 - 376 line/column: 20:2 - 20:4
[LEFT_CURLY]	{		index: 378 - 378 line/column: 20:6 - 20:6
[WHILE]	while		index: 383 - 387 line/column: 21:3 - 21:7
[LEFT_PAREN]	(		index: 388 - 388 line/column: 21:8 - 21:8
[IDENTIFIER]	b		index: 389 - 389 line/column: 21:9 - 21:9
[RIGHT_PAREN]	)		index: 390 - 390 line/column: 21:10 - 21:10
[LEFT_CURLY]	{		index: 391 - 391 line/column: 21:11 - 21:11
[DO]	do		index: 397 - 398 line/column: 22:4 - 22:5
[LEFT_CURLY]	{		index: 399 - 399 line/column: 22:6 - 22:6
[BREAK]	break		index: 406 - 410 line/column: 23:5 - 23:9
[SEMICOLON]	;		index: 411 - 411 line/column: 23:10 - 23:10
[RIGHT_CURLY]	}		index: 417 - 417 line/column: 24:4 - 24:4
[WHILE]	while		index: 419 - 423 line/column: 24:6 - 24:10
[LEFT_PAREN]	(		index: 425 - 425 line/column: 24:12 - 24:12
[FALSE]	false		index: 426 - 430 line/column: 24:13 - 24:17
[RIGHT_PAREN]	)		index: 431 - 431 line/column: 24:18 - 24:18
[SEMICOLON]	;		index: 432 - 432 line/column: 24:19 - 24:19
[CONTINUE]	continue		index: 438 - 445 line/column: 25:4 - 25:11
[SEMICOLON]	;		index: 446 - 446 line/column: 25:12 - 25:12
[RIGHT_CURLY]	}		index: 451 - 451 line/column: 26:3 - 26:3
[RIGHT_CURLY]	}		index: 458 - 458 line/column: 27:2 - 27:2
[CATCH]	catch		index: 463 - 467 line/column: 28:2 - 28:6
[LEFT_PAREN]	(		index: 469 - 469 line/column: 28:8 - 28:8
[IDENTIFIER]	Exception		index: 470 - 478 line/column: 28:9 - 28:17
[IDENTIFIER]	e		index: 480 - 480 line/column: 28:19 - 28:19
[RIGHT_PAREN]	)		index: 481 - 481 line/column: 28:20 - 28:20
[LEFT_CURLY]	{		index: 483 - 483 line/column: 28:22 - 28:22
[RIGHT_CURLY]	}		index: 485 - 485 line/column: 28:24 - 28:24
[FINALLY]	finally		index: 490 - 496 line/column: 29:2 - 29:8
[LEFT_CURLY]	{		index: 498 - 498 line/column: 29:10 - 29:10
[RIGHT_CURLY]	}		index: 500 - 500 line/column: 29:12 - 29:12
[RETURN]	return		index: 504 - 509 line/column: 30:2 - 30:7
[NUMBER]	1		index: 511 - 511 line/column: 30:9 - 30:9
[SEMICOLON]	;		index: 512 - 512 line/column: 30:10 - 30:10
[RIGHT_CURLY]	}		index: 515 - 515 line/column: 31:1 - 31:1
[RIGHT_CURLY]	}		index: 517 - 517 line/column: 32:0 - 32:0